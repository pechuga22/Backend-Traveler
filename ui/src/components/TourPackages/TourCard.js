import React from 'react'
import { Image, Rating, Icon, Card, Grid } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

const TourCard = ({ tour }) => (
  <Grid.Column id={tour._id}>
    <Link to={`/tours/${tour._id}`}>
      <Image src={tour.image} size='small' centered alt={tour.location} />
      <Card.Content>
        <Card.Header className='card title'>{tour.title}</Card.Header>
        <Grid.Column>
          <Card.Meta>
            <Icon name='map marker alternate' />
            <span>{tour.location}</span>
          </Card.Meta>
          <Card.Content>
            <Rating icon='favorite' defaultRating={tour.rating} maxRating={5} />
            <span className='rate'>{tour.rating}</span>
            <span className='counter'>({tour.ratingCount})</span>
          </Card.Content>
        </Grid.Column>
      </Card.Content>
    </Link>
  </Grid.Column>
)

export default TourCard
