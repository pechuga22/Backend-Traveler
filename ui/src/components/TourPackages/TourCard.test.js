import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TourCard from './TourCard'

const tour = {
    _id: '3402397a-01a6-4bad-87a8-cdc36e4e6c0c',
    title: 'El mejor recorrido en Bici hasta Incachaca',
  }

describe('./components/TourPackages/TourCard.js', () => {
  describe('.render()', () => {
    it('should render the TourCard component', () => {
      const component = shallow(<TourCard tour={tour} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('find the rate class into TourCard component', () => {
      const component = shallow(<TourCard tour={tour} />)
      const wrapper = component.find('.rate')
      expect(wrapper.length).toBe(1)
    })
  })
})
