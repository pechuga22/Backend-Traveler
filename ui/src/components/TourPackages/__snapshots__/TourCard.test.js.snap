// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/TourPackages/TourCard.js .render() should render the TourCard component 1`] = `
<GridColumn
  id="3402397a-01a6-4bad-87a8-cdc36e4e6c0c"
>
  <Link
    to="/tours/3402397a-01a6-4bad-87a8-cdc36e4e6c0c"
  >
    <Image
      as="img"
      centered={true}
      size="small"
      ui={true}
    />
    <CardContent>
      <CardHeader
        className="card title"
      >
        El mejor recorrido en Bici hasta Incachaca
      </CardHeader>
      <GridColumn>
        <CardMeta>
          <Icon
            as="i"
            name="map marker alternate"
          />
          <span />
        </CardMeta>
        <CardContent>
          <Rating
            clearable="auto"
            icon="favorite"
            maxRating={5}
          />
          <span
            className="rate"
          />
          <span
            className="counter"
          >
            (
            )
          </span>
        </CardContent>
      </GridColumn>
    </CardContent>
  </Link>
</GridColumn>
`;
